samples/subsys/usbc/sink/src/main.c

```diff
@@ -0,0 +1,382 @@
+{
+	struct port1_data_t *dpm_data = usbc_get_dpm_data(dev);
+
+	return build_request_data_object(dpm_data);
+}
+/* usbc.rst callbacks end */
+
+/* usbc.rst notify start */
+/**
+ * @brief Port Policy Callback to notify the application of an event that occurred on this port
+ */
+static void usbc_port1_notify(const struct device *dev, enum policy_notify_t policy_notify)
+{
+	switch (policy_notify) {
+	case PROTOCOL_ERROR:
+		break;
+	case MSG_DISCARDED:
+		break;
+	case MSG_ACCEPT_RECEIVED:
+		break;
+	case MSG_REJECTED_RECEIVED:
+		break;
+	case MSG_NOT_SUPPORTED_RECEIVED:
+		break;
+	case TRANSITION_PS:
+		/* Display the Source Capabilities */
+		display_source_caps(dev);
+		break;
+	case PD_CONNECTED:
+		break;
+	case NOT_PD_CONNECTED:
+		break;
+	case POWER_CHANGE_0A0:
+		LOG_INF("PWR 0A\n");
+		break;
+	case POWER_CHANGE_DEF:
+		LOG_INF("PWR DEF\n");
+		break;
+	case POWER_CHANGE_1A5:
+		LOG_INF("PWR 1A5\n");
+		break;
+	case POWER_CHANGE_3A0:
+		LOG_INF("PWR 3A0\n");
+		break;
+	case DATA_ROLE_IS_UFP:
+		break;
+	case DATA_ROLE_IS_DFP:
+		break;
+	case PORT_PARTNER_NOT_RESPONSIVE:
+		LOG_INF("Port Partner not PD Capable\n");
+		break;
+	case SNK_TRANSITION_TO_DEFAULT:
+		break;
+	case HARD_RESET_RECEIVED:
+		break;
+	}
+}
+/* usbc.rst notify end */
+
+/* usbc.rst check start */
+/**
+ * @brief Port Policy Callback to check if the USBC subsystem should take an action
+ */
+bool usbc_port1_policy_check(const struct device *dev, enum policy_check_t policy_check)
+{
+	switch (policy_check) {
+	case CHECK_POWER_ROLE_SWAP:
+		/* Reject power role swaps */
+		return false;
+	case CHECK_DATA_ROLE_SWAP_TO_DFP:
+		/* Reject data role swap to DFP */
+		return false;
+	case CHECK_DATA_ROLE_SWAP_TO_UFP:
+		/* Reject data role swap to UFP */
+		return true;
+	case CHECK_SNK_AT_DEFAULT_LEVEL:
+		/* This device is always at the default power level */
+		return true;
+	default:
+		/* Reject all other policy checks */
+		return false;
+
+	}
+}
+/* usbc.rst check end */
+
+/* usbc.rst init sink caps start */
+/**
+ * @brief Initialize this port's Sink Capabilities
+ */
+static void init_port1_snk_caps(void)
+{
+	union pd_fixed_supply_pdo_sink fixed_supply_pdo_sink;
+
+	fixed_supply_pdo_sink.operational_current = PD_CONVERT_MA_TO_FIXED_PDO_CURRENT(100);
+	fixed_supply_pdo_sink.voltage = PD_CONVERT_MV_TO_FIXED_PDO_VOLTAGE(5000);
+	fixed_supply_pdo_sink.frs_required = FRS_NOT_SUPPORTED;
+	fixed_supply_pdo_sink.dual_role_data = false;
+	fixed_supply_pdo_sink.usb_comms_capable = false;
+	fixed_supply_pdo_sink.unconstrained_power = false;
+	fixed_supply_pdo_sink.higher_capability = false;
+	fixed_supply_pdo_sink.dual_role_power = false;
+	fixed_supply_pdo_sink.type = PDO_FIXED;
+
+	port1_data.snk_caps[0] = fixed_supply_pdo_sink.raw_value;
+	port1_data.snk_cap_cnt = 1;
+}
+/* usbc.rst init sink caps end */
+
+void main(void)
+{
+	const struct device *usbc_port1;
+	int ret;
+
+	/* usbc.rst config board start */
+	/* Config the board */
+	ret = board_config();
```

Comment:
No way, let's not go down the road of board specific C files called into by main like this. This is the sort of thing that should happen before main should even run, using a SYS_INIT or similar defined by the driver or subsystem layer.

Reply:
I agree and this has been updated